invoice/src/main/java/org/killbill/billing/invoice/generator/FixedAndRecurringInvoiceItemGenerator.java

```diff
@@ -272,11 +275,16 @@ boolean isSameDayAndSameSubscription(final InvoiceItem prevComputedFixedItem, fi
                         }
                     }
                     updatePerSubscriptionNextNotificationDate(thisEvent.getSubscriptionId(), itemDataWithNextBillingCycleDate.getNextBillingCycleDate(), items, recurringBillingMode,
-                                                              perSubscriptionFutureNotificationDate);
+                                                              perSubscriptionFutureNotificationDates);
+                }
+            } else {
+                final SubscriptionFutureNotificationDates futureNotificationDates = perSubscriptionFutureNotificationDates.get(thisEvent.getSubscriptionId());
```

Comment:
Is my understanding correct that prior to this change, a subscription with a blocked billing period might have had a (NULL) invoice run after the paused date?

Reply:
I think this is true. I ''ll verify the state of our pause/resume tests for `in-advance` and add one for `in-arrear`.